pilot/pkg/security/authz/matcher/header.go

```diff
@@ -55,6 +57,37 @@ func HeaderMatcher(k, v string) *routepb.HeaderMatcher {
 	}
 }
 
+// HeaderMatcherWithRegex converts a key, value string pair to a corresponding
+// HeaderMatcher to support matching the value presented in the inline header with
+// multiple values concatenated by commas, as per RFC7230.
+func HeaderMatcherWithRegex(k, v string) *routepb.HeaderMatcher {
```

Comment:
I'm not certain that this is correct since it seems like the semantics of `,` depend on the header field. For example, if `k = "Last-Modified"`, `v  = Sat`, then a request with `Last-Modified: Sat, 04 May 1996` matches, when it shouldn't.

Reply:
This is called out in the [security note](https://github.com/istio/istio/pull/56085/files#diff-a3e7caa9a4e9e45d7a8b1fbebe4acac954bbf9faf8ba0ec7b0e2789fadcdba44R26-R29) that, if the header you are matching for is not obeying the inline header format as per RFC7230, then you shouldn't use this header matcher.

Reply:
Oh I see this isn't for all matching, sorry for the misunderstanding.